pkg/config/config.go

```diff
@@ -30,9 +32,25 @@ type Info interface {
 type ComponentSettings struct {
 
 	// The builder image to use
-	ComponentType *string `yaml:"ComponentType,omitempty"`
+	Type *string `yaml:"Type,omitempty"`
+
+	// Path is path to binary in current/context dir
+	Path *string `yaml:"Path,omitempty"`
+
+	// Ref is component source git ref but can be levaraged for more in future
+	Ref *string `yaml:"Ref,omitempty"`
```

Comment:
here we are storing `Ref` but where would be store url to the git repository?

Reply:
In Path

Reply:
Again, the comment doesn't say that. Also in that case, `Path` is not a good name for it. If it can be git url we have to find a different name for it.

Reply:
Would SourceLocation be better name? @kadel

Reply:
> Would SourceLocation be better name? @kadel

that is better

Reply:
so just to confirm change `Path` to `SourceLocation`?

Reply:
`SourceLocation` still sounds a bit weird, but I can't think of anything better right now. It still makes more sense than just `Path`. Also please explain what this is used for in the comment.

Reply:
@anmolbabu what is `SourceLocation` used for?

Reply:
added comment

Reply:
And renamed the field

Reply:
> Also please explain what this is used for in the comment

As mentioned in https://github.com/redhat-developer/odo/pull/1453/files#r265439578 `SourceLocation` can also bit git url, comment needs to say that

Reply:
added @kadel